### ISSUE TITLE
Issues with "-B" != 38

### ISSUE THREAD

**Issue by lars18th** [gfto/mptsd#18]
Hi,

When using the parameter "-B" for streaming with different bandwith than the default one (38Mbps), after some time of starting the execution I see error messages and the output stream corrupts:

```
Jun 15 00:39:18 host ux-mptsd: INPUT : [ch1         ] Disable input, buffer is full. fd: 4 src: udp://239.0.0.1:12345
```

I feel the INPUT buffer goes full because some timing in the code doesn't work for _less_ than the stock 38Mbps. Any idea about how to fix it?

Thank you!

**Comment by lars18th** [comment]
Hi,

I feel this issue is related to inputs with padding (NULL packets). I resolved it with this PR #22 .

I hope you merge it! :smile:

**Comment by lars18th** [comment]
Hi,

I found the bug!
This PR #23 solves the trouble.
Now you can use any output bitrate without troubles! :wink:

**Comment by lars18th** [comment]
Hi,

The PR #27 solves this problem entirely! :smile: 
I hope it be merged soon. :crossed_fingers:

**Comment by salvaaod** [comment]
Hi lars18th, does #28 has the #27 fix included ?

**Comment by lars18th** [comment]
> Hi lars18th, does #28 has the #27 fix included ?

No. They (#28 and #27) are _independent_ patches. So you need to apply all!

Regards.

**Comment by salvaaod** [comment]
> > Hi lars18th, does #28 has the #27 fix included ?
> 
> No. They (#28 and #27) are _independent_ patches. So you need to apply all!
> 
> Regards.

Thanks, and I understand that #27 supersedes #23 and #22  or should I apply all to get rid of the null padding with low bitrates ?
